server/src/main/java/io/crate/execution/dml/IndexDocumentBuilder.java

```diff
@@ -190,6 +192,9 @@ public ParsedDocument build(String id) {
 
         BytesRef idBytes = Uid.encodeId(id);
         addField(new Field(SysColumns.Names.ID, idBytes, SysColumns.ID.FIELD_TYPE));
+        if (tableVersionCreated.onOrAfter(IdCollectorExpression.STORED_AS_BINARY_VERSION)) {
+            addField(new BinaryDocValuesField(SysColumns.Names.ID, idBytes));
+        }
```

Comment:
Is it expected that on >6.0 we end up having 2 fields with name SysColumns.Names.ID?
Regular `Field` and a `BinaryDocValuesField`

Reply:
It's still just a single field: in lucene you can add multiple Field instances to a Document, and the IndexWriter will gather them all together.